### ISSUE TITLE
(7.2) SimPermut displays no item options, throws LUA error attempt to concatenate '?' (nil value)

### ISSUE THREAD

**Issue by CountVonn** [SimCMinMax/SimPermut#1]
In 7.2, SimPermut is no longer presenting items or checkboxes on some of my characters.  An LUA error appears instead; see below for the full error message.  This has affected one of my characters that has quest completed for the Power Unbound (which modifies the artifact weapon).  SimPermut is still working correctly on an alt that has not completed that quest line.  Item IDs imported by Simcraft are also pasted below as it seems like it could be relevant, from both a working and non-working character.

Full LUA error:
```
Message: Interface\AddOns\SimPermut\core.lua:615: attempt to concatenate field '?' (a nil value)
Time: 03/28/17 21:17:14
Count: 1
Stack: Interface\AddOns\SimPermut\core.lua:615: in function `GetItemString'
Interface\AddOns\SimPermut\core.lua:1230: in function `GetBaseString'
Interface\AddOns\SimPermut\core.lua:307: in function `BuildFrame'
Interface\AddOns\SimPermut\core.lua:103: in function `?'
Interface\FrameXML\ChatFrame.lua:4397: in function `ChatEdit_ParseText'
Interface\FrameXML\ChatFrame.lua:4065: in function `ChatEdit_SendText'
Interface\FrameXML\ChatFrame.lua:4101: in function `ChatEdit_OnEnterPressed'
[string "*:OnEnterPressed"]:1: in function <[string "*:OnEnterPressed"]:1>

Locals: self = <table> {
 SetDefaultModuleLibraries = <function> defined @Interface\AddOns\SimPermut\libs\AceAddon-3.0\AceAddon-3.0.lua:398
 statsStringCorres = <table> {
 }
 EnableModule = <function> defined @Interface\AddOns\SimPermut\libs\AceAddon-3.0\AceAddon-3.0.lua:363
 CheckItemCount = <function> defined @Interface\AddOns\SimPermut\core.lua:465
 baseName = "SimPermut"
 BuildItemFrame = <function> defined @Interface\AddOns\SimPermut\core.lua:315
 RelicSlots = <table> {
 }
 Printf = <function> defined @Interface\AddOns\SimPermut\libs\AceConsole-3.0\AceConsole-3.0.lua:69
 SetDefaultModulePrototype = <function> defined @Interface\AddOns\SimPermut\libs\AceAddon-3.0\AceAddon-3.0.lua:440
 enchantRing = <table> {
 }
 GetAllPermutations = <function> defined @Interface\AddOns\SimPermut\core.lua:930
 defaultModuleState = true
 IsEnabled = <function> defined @Interface\AddOns\SimPermut\libs\AceAddon-3.0\AceAddon-3.0.lua:482
 ProfNames = <table> {
 }
 DisableModule = <function> defined @Interface\AddOns\SimPermut\libs\AceAddon-3.0\AceAddon-3.0.lua:381
 enchantCloak = <table> {
 }
 PermutSlotNames = <table> {
 }
 RegisterMessage = <function> defined @Interface\AddOns\DBM-Core\Libs\CallbackHandler-1.0\CallbackHandler-1.0.lua:116
 ArtifactTable = <table> {
 }
 GetFinalString = <function> defined @Interface\AddOns\SimPermut\core.lua:1253
 GetListItem = <function> defined @Interface\AddOns\SimPermut\core.lua:721
 CheckUsability = <function> defined @Interface\AddOns\SimPermut\core.lua:1258
 SetEnabledState = <function> defined @Interface\AddOns\SimPermut\libs\AceAddon-3.0\AceAddon-3.0.lua:455
 gemList = <table> {
 }
 enabledState = true
 RegisterEvent = <function> defined @Interface\AddOns\DBM-Core\Libs\CallbackHandler-1.0\CallbackHandler-1.0.lua:116
 SpecNames = <table> {
 }
 GetItemString = <function> defined @Interface\AddOns\SimPermut\core.lua:491
 IterateModules = <function> defined @Interface\AddOns\SimPermut\libs\AceAddon-3.0\AceAddon-3.0.lua:468
 ReorganizeEquip = <function> defined @Interface\AddOns\SimPermut\core.lua:942
 PreparePermutations = <function> defined @Interface\AddOns\SimPermut\core.lua:966
 OnInitialize = <function> defined @Interface\AddOns\SimPermut\core.lua:118
 RegisterChatCommand = <function> defined @Interface\AddOns\SimPermut\libs\AceConsole-3.0\AceConsole-3.0.lua:85
 GetModule = <function> defined @Interface\AddOns\SimPermut\libs\AceAddon-3.0\AceAddon-3.0.lua:241
 enchantNeck = <table> {
 }
 defaultModuleLibraries = <table> {
 }
 UnregisterChatCommand = <function> defined @Interface\AddOns\SimPermut\libs\AceConsole-3.0\AceConsole-3.0.lua:111
 GetCopyName = <function> defined @Interface\AddOns\SimPermut\core.lua:1129
 GetArtifactString = <function> defined @Interface\AddOns\SimPermut\core.lua:1327
 BuildFrame = <function> defined @Interface\AddOns\SimPermut\core.lua:133
 GetName = <function> defined @Interface\AddOns\SimPermut\libs\AceAddon-3.0\AceAddon-3.0.lua:310
 GetItemStrings = <function> defined @Interface\AddOns\SimPermut\core.lua:681
 name = "SimPermut"
 GetBaseString = <function> defined @Interface\AddOns\SimPermut\core.lua:1169
 orderedModules = <table> {
 }
 slotNames = <table> {
 }
 Disable = <function> defined @Interface\AddOns\SimPermut\libs\AceAddon-3.0\AceAddon-3.0.lua:345
 GetPermutationString = <function> def
```

Currently equipped items IDs from non-working character:
```
deathknight="Vôn"
origin="https://us.api.battle.net/wow/character/lightbringer/Vôn/advanced"
thumbnail="http://us.battle.net/static-render/us/lightbringer/221/131310301-avatar.jpg"
level=110
race=dwarf
role=attack
position=back
professions=jewelcrafting=790/mining=791
talents=http://us.battle.net/wow/en/tool/talent-calculator#dZ!2212121
artifact=12:0:0:0:0:108:3:109:3:110:3:111:3:113:3:114:3:115:3:117:3:119:1:120:1:122:1:123:1:124:1:1090:3:1091:1:1092:1:1332:1:1360:1
spec=frost

head=dreadwyrm_crown,id=138355,bonus_id=3514/1507/3337
neck=prydaz_xavarics_magnum_opus,id=132444,bonus_id=3458/3455/1811,gems=200str,enchant=mark_of_the_claw
shoulders=mortar_guard_shoulderplates,id=137524,bonus_id=3536/1552/3337
back=dreadwyrm_greatcloak,id=138364,bonus_id=3514/1482/3336,enchant=200str
chest=dreadwyrm_breastplate,id=138349,bonus_id=3514/43/1472/1813
shirt=stylish_black_shirt,id=3427
tabard=renowned_guild_tabard,id=69210
wrists=deepfurrow_bracers,id=137337,bonus_id=3536/1552/3337
hands=dreadwyrm_gauntlets,id=138352,bonus_id=3520/1808/1457,gems=150haste
waist=koltiras_newfound_will,id=132366,bonus_id=3529/3530/1811
legs=tassets_of_perpetual_despair,id=142127,bonus_id=3418/42/1517/3337
feet=sabatons_of_unchanging_fate,id=140891,bonus_id=3514/1487/3337
finger1=band_of_twisted_bark,id=134531,bonus_id=3418/1537/3337,enchant=200haste
finger2=jeweled_signet_of_melandrus,id=134542,bonus_id=1727/1808/1507/3337,gems=150haste,enchant=200haste
trinket1=convergence_of_fates,id=140806,bonus_id=3514/1477/3336
trinket2=unstable_arcanocrystal,id=141482,bonus_id=3466/1472
main_hand=blades_of_the_fallen_prince,id=128292,bonus_id=717,gem_id=137308/140821/137340/0,relic_id=3417:1527:3337/3443:1467:1813/3415:1522:3337/0,enchant=rune_of_razorice
off_hand=blades_of_the_fallen_prince,id=128293,enchant=rune_of_the_fallen_crusader
```

Item IDs from a working character:
```
paladin="Dulam"
origin="https://us.api.battle.net/wow/character/lightbringer/Dulam/advanced"
thumbnail="http://us.battle.net/static-render/us/lightbringer/135/110805383-avatar.jpg"
level=110
race=dwarf
role=attack
position=back
professions=herbalism=508/mining=423
talents=http://us.battle.net/wow/en/tool/talent-calculator#bb!0021111
artifact=2:0:0:0:0:40:1:41:3:42:3:43:3:44:3:47:3:49:1:50:3:51:3:53:3:54:1:350:1:351:1:353:1:1275:1
spec=retribution

head=manacrystaladorned_helmet,id=141459,bonus_id=3466/1472
neck=belerons_choker_of_misery,id=140899,bonus_id=3516/1487/1813
shoulders=shoulderplates_of_oversized_sorcery,id=141426,bonus_id=40/3466/1472
back=greatmantle_of_the_highlord,id=138369,bonus_id=3520/1487/3337
chest=polished_jade_chestguard,id=137527,bonus_id=3415/1522/3337
tabard=renowned_guild_tabard,id=69210
wrists=uthers_guard,id=137105,bonus_id=3529/3530
hands=rockbound_gauntlets,id=134136,bonus_id=3474/1512/3336
waist=slack_tide_girdle,id=133770,bonus_id=3415/1522/3337
legs=tassets_of_perpetual_despair,id=142127,bonus_id=3453/1482/3336
feet=aggramars_stride,id=132443,bonus_id=3529/3530
finger1=ring_of_exclusive_servitude,id=140906,bonus_id=3520/1808/1457
finger2=band_of_callous_dominance,id=134528,bonus_id=3413/1517/3337
trinket1=memento_of_angerboda,id=133644,bonus_id=3416/1512/1813
trinket2=brulstone_idol,id=134146,bonus_id=3474/607/1512/3336
main_hand=ashbringer,id=120978,bonus_id=737,gem_id=143695/140043/141260/0,relic_id=3432:1497:1674/3432:1522:3337/3432:1497:1674/0
```

![dulam_lightbringer_working](https://cloud.githubusercontent.com/assets/4949110/24434468/c2894844-13fd-11e7-903b-6687c0e4756e.jpg)
![von_lightbringer_lua_error](https://cloud.githubusercontent.com/assets/4949110/24434470/c470c3a8-13fd-11e7-8b49-6954866e126c.jpg)

**Comment by Kutikuti** [comment]
The issue happened only after completing the new artifact quest to unlock new traits right ?
Haven't completed it yet so I can't see any errors. I will try to complete it but it won't be until tonight.


By any chance, do you have simulationcraft addon installed ?
If yes, could you do a /simc and see if the problem is here too ?
I'm in touch with the dev and this is a pretty big issue if it happens with this addon too

Anyway, thanks for the report

**Comment by Kutikuti** [comment]
I did the quest and the issue is also in simc addon. I will see what i can do

**Comment by Kutikuti** [comment]
It should be fixed in 1.8.1 currently under review by curse

**Comment by CountVonn** [comment]
I pulled 1.8.1 from the repo and updated my local addon folder manually.  The change looks good, I'm no longer seeing the LUA error when running /simpermut on my character that has completed the new artifact quest line.  I did some quick tests of functionality and both Generate and AutoSimc Export buttons are working correctly.  Thanks for the quick response, much appreciated!
